addons/html_builder/static/src/builder/options/animate_option.js

```diff
@@ -13,27 +15,316 @@ class AnimateOptionPlugin extends Plugin {
+                        "o_visible",
+                        "o_animated",
+                        "o_animate_out"
+                    );
+                    editingElement.style.animationDelay = "";
+                    editingElement.style.animationPlayState = "";
+                    editingElement.style.animationName = "";
+                    editingElement.style.visibility = "";
+
+                    if (effectName === "onScroll") {
+                        delete editingElement.dataset.scrollZoneStart;
+                        delete editingElement.dataset.scrollZoneEnd;
+                    }
+                    if (effectName === "onHover") {
+                        // todo: to implement
+                        // this.trigger_up("option_update", {
+                        //     optionName: "ImageTools",
+                        //     name: "disable_hover_effect",
+                        // });
+                    }
+
+                    const isNextEffectFadein = effectWithFadein.includes(nextAction.value);
+                    if (!isNextEffectFadein) {
+                        this.removeEffectAndDirectionClasses(
+                            dependencyManager,
+                            editingElement.classList
+                        );
+                        editingElement.style.setProperty("--wanim-intensity", "");
+                        editingElement.style.animationDuration = "";
+                        this.setImagesLazyLoading(editingElement, true);
+                    }
+                },
+                apply: ({ editingElement, value: effectName }) => {
+                    if (effectWithFadein.includes(effectName)) {
+                        editingElement.classList.add("o_anim_fade_in");
+                        this.setImagesLazyLoading(editingElement, false);
+                    }
+                    if (effectName === "onScroll") {
+                        editingElement.dataset.scrollZoneStart = 0;
+                        editingElement.dataset.scrollZoneEnd = 100;
+                    }
+                    if (effectName === "onHover") {
+                        // todo: to implement
+                        // Pause the history until the hover effect is applied in
+                        // "setImgShapeHoverEffect". This prevents saving the intermediate
+                        // steps done (in a tricky way) up to that point.
+                        // this.options.wysiwyg.odooEditor.historyPauseSteps();
+                        // this.trigger_up("option_update", {
+                        //     optionName: "ImageTools",
+                        //     name: "enable_hover_effect",
+                        // });
+                    }
+                },
+            },
+            setAnimateIntensity: {
+                getValue: ({ editingElement }) => {
+                    const intensity = parseInt(
+                        window
```

Comment:
this.window ?

Reply:
What do you mean? There is no window on the plugin. You mean `this.document.defaultView`?

Reply:
yes